src/libcore/iter.rs

```diff
@@ -280,6 +289,16 @@ pub trait Iterator {
     /// different sized integer, the `zip` function provides similar
     /// functionality.
     ///
+    /// # Undefined overflow
+    ///
+    /// The method does no guarding against overflows, so enumerating more than
+    /// `usize::MAX` elements is undefined.
```

Comment:
@huonw AIUI the sentence says that the common operations produce the same results as before if they're in the domain of the target type.

>Change the semantics of the built-in fixed-size integer types from being defined as wrapping around on overflow to it being considered a program error (but not undefined behavior in the C sense).

We can certainly change "undefined" to "unspecified" in the doc here, but I think the RFC *does not* specify that wrap-on-overflow is guaranteed, but rather any value may be returned.